lang/tags/functions.py

```diff
@@ -19,6 +25,12 @@ def code_type(m) -> str:
     return m.code_type
 
 
+@mod.capture(rule="{user.code_function_modifiers}+")
+def code_function_modifiers(m) -> str:
+    """Returns a space separated string of function modifiers"""
+    return " ".join(m)
```

Comment:
Why join this with strings if you're just going to split it? Could just inline the above rule into the talon file and then use `_list` suffix to get an array

Reply:
I wanted to have a capture rather than a raw list as they allow for more flexible overriding. Then that meant I had to split out the no modifiers command because you can't make the capture rule optional. And in .talon you can't construct an empty list. So then I could have the type be e.g. `Union[List[str], int]`, but I figured just using str like this was fine.

That all said, I can't really think of a reason somebody would want the flexibility of a capture, so I might just go to a raw list like you said :).